lib/resourcedelete/batch.go

```diff
@@ -37,3 +37,32 @@ func DeleteJobv1(ctx context.Context, client batchclientv1.JobsGetter, required
 	}
 	return true, nil
 }
+
+// DeleteCronJobv1 checks the given resource for a valid delete annotation. If found
+// it checks the status of a previously issued delete request. If delete has not been
+// requested and in UpdatingMode it will issue a delete request.
+func DeleteCronJobv1(ctx context.Context, client batchclientv1.CronJobsGetter, required *batchv1.CronJob,
+	updateMode bool) (bool, error) {
+
+	if delAnnoFound, err := ValidDeleteAnnotation(required.Annotations); !delAnnoFound || err != nil {
+		return delAnnoFound, err
+	}
+	existing, err := client.CronJobs(required.Namespace).Get(ctx, required.Name, metav1.GetOptions{})
```

Comment:
Hmmm is it right to entirely ignore errors here? We immediately overwrite it on L56 and we end up using `existing` on L62. Either we should not be ignoring the error, or we should do so explicitly with a comment why (`existing, _ := ...`)

Reply:
It took me few a seconds but the initial `err` does get passed to the function `GetDeleteProgress()` on the L56 and thus is not immediately overwritten:

```go
	if deleteRequested, err := GetDeleteProgress(resource, err); err == nil {
```

And there it is used quite a few times:

https://github.com/openshift/cluster-version-operator/blob/master/lib/resourcedelete/helper.go#L100-L132

Reply:
Sorry, I forgot about your new comment when pushing the new changes. However, the new code structure simplifies the logic in my opinion. 

https://github.com/Davoska/cluster-version-operator/blob/bug-2039228-cvo-hotloops-on-cronjob/lib/resourcedelete/batch.go#L43-L66

```go
	if delAnnoFound, err := ValidDeleteAnnotation(required.Annotations); !delAnnoFound || err != nil {
		return delAnnoFound, err
	}
	existing, err := client.CronJobs(required.Namespace).Get(ctx, required.Name, metav1.GetOptions{})
	resource := Resource{
		Kind:      "cronjob",
		Namespace: required.Namespace,
		Name:      required.Name,
	}
	deleteRequested, err := GetDeleteProgress(resource, err)
	if err != nil {
		return true, fmt.Errorf("Error running delete for %s, err=%v", resource, err)
	}
	if !deleteRequested && updateMode {
		if err := client.CronJobs(required.Namespace).Delete(ctx, required.Name, metav1.DeleteOptions{}); err != nil {
			return true, fmt.Errorf("Delete request for %s failed, err=%v", resource, err)
		}
		SetDeleteRequested(existing, resource)
	}
```

Now it's clearer than before, and something like `getErr` is no longer needed IMO. However, if you feel like it should be there even now, I am happy to add it.

Reply:
It's fine, no more modifications needed, thanks!